### ISSUE TITLE
ModelConfig fidelity_weight. A Statistical Fidelity and Outlier Control

### ISSUE THREAD

**Issue by AishaJaff** [dataxid/dataxid-python#44]
## Problem

DataXID currently operates as a black box with no way to control how closely synthetic output matches the original distribution. This creates two irreconcilable problems for different use cases:

**Problem 1 - No user control over distribution fidelity.**

The SDK learns statistical distributions during training but provides no way for developers to control how closely the synthetic output should match the original. Two common use cases require opposite behavior:

A fraud detection team needs high fidelity, which means the synthetic data must preserve rare fraud patterns that represent only 0.1% of transactions. If the model smooths these away, the synthetic data cannot train a useful fraud classifier.

A data sharing team needs high generalization, which means the synthetic data must diverge enough from the original to prevent re-identification. If the model is too faithful, rare combinations of values can identify individuals even without any single PII field.

Currently, both teams use the same default settings with no way to adjust.

**Problem 2 - Overfitting: The privacy leak.**

When `privacy_enabled=True` is set, a developer expects stronger privacy guarantees. However there is no mechanism to prevent the model from memorizing rare combinations of values that effectively identify individuals. A synthetic row combining `age=87`, `income=3,200,000`, and `postcode=SW1A` forexample may not contain any single identifying field but the combination uniquely identifies the training record.

The current `privacy_noise parameter` only adds Gaussian noise to embeddings. It does not control whether the generative model learns rare distributional patterns. A high-fidelity model reproduces these patterns regardless of noise level.

## Proposed Solution

Introduce `fidelity_weight`, `smoothing`, and `clip_outliers` parameters in `ModelConfig` that let developers position their synthesis on the fidelity-privacy spectrum:

Scenario 1: ML training to maximize statistical fidelity

```python
import dataxid

# Preserves rare values and exact distributions for downstream model training
config_ml = dataxid.ModelConfig(
    fidelity_weight=0.95,   # Stay close to original distribution
    smoothing=0.05,          # Minimal smoothing of rare categories
    clip_outliers=False      # Preserve outliers which are needed for anomaly detection
)
syn_ml = dataxid.synthesize(data=df, n_samples=1000, config=config_ml)

```
Scenario 2: Public data release to maximize privacy

```python
# Generalizes distribution to prevent individual identification
config_safe = dataxid.ModelConfig(
    fidelity_weight=0.3,    # Allow significant divergence from original
    smoothing=0.7,           # Smooth rare categories into plausible neighbors
    clip_outliers=True       # Removes statistical outliers that could identify individuals
)
syn_safe = dataxid.synthesize(data=df, n_samples=1000, config=config_safe)

```

**What smoothing would fix in the current pipeline:**
Instead of the current binary choice (keep a category exactly or replace with `_RARE_`), a smoothing layer would generate statistically plausible neighboring values, preserving the shape of the distribution without memorizing specific training records.

The model already uses `label_smoothing` in the `training config`, passed to` init-training` in `training/_frozen.py`. The proposed `fidelity_weight` extends this concept from the loss function to the generation step, controlling the temperature of categorical sampling at inference time, giving developers explicit control over the fidelity-privacy tradeoff.

## Alternatives Considered

1. Differential privacy `(privacy_enabled=True)`: DP adds noise to embeddings, addressing the privacy boundary between local processing and the server. It does not control the fidelity of the distribution learned by the server-side model. A developer cannot use DP to fix the ML accuracy gap or the mode collapse problem.
2. Manual post-processing: Developers can attempt to smooth synthetic data after generation using custom scripts. However once the model has collapsed rare categories to `_RARE_`, the information is gone and no post-processing can recover what the model failed to learn. Smoothing must happen during encoding or generation, not after.
3. Conditional generation (Issue #7): Conditional generation controls which rows get generated based on output constraints. Fidelity tuning controls how faithfully the model learns the input distribution. These are complementary features, a developer might use both together: high fidelity to learn a rare fraud pattern, then conditional generation to oversample that pattern. 
4. Generating more samples and filtering: Larger `n_samples` does not help with mode collapse. If a value pattern was dropped during analysis, the generative model never learned it and cannot produce it regardless of how many rows are requested.
